flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/FlinkInputFormat.java

```diff
@@ -77,13 +79,13 @@ public BaseStatistics getStatistics(BaseStatistics cachedStatistics) {
     tableLoader.open();
     try (TableLoader loader = tableLoader) {
       Table table = loader.loadTable();
-      return FlinkSplitGenerator.createInputSplits(table, context);
+      return FlinkSplitGenerator.createInputSplits(table, context, localityPreferred);
     }
   }
 
   @Override
   public InputSplitAssigner getInputSplitAssigner(FlinkInputSplit[] inputSplits) {
-    return new DefaultInputSplitAssigner(inputSplits);
+    return new LocatableInputSplitAssigner(inputSplits);
```

Comment:
Shouldn't this check `localityPreferred` and return a `DefaultInputSplitAssigner` if it is false? That way we get the same behavior for all file stores other than HDFS.